tests/numpy/test_gen_data.py

```diff
@@ -101,3 +103,54 @@ def test_can_create_arrays_of_tuples():
         lambda x: all(t[0] != t[1] for t in x))
     for a in arr:
         assert a in ((1, 0), (0, 1))
+
+
+@given(array_shapes())
+def test_can_generate_array_shapes(shape):
+    assert isinstance(shape, tuple)
+    assert all(isinstance(i, int) for i in shape)
+
+
+@given(st.integers(1, 10), st.integers(0, 9), st.integers(1), st.integers(0))
+def test_minimise_array_shapes(min_dims, dim_range, min_side, side_range):
+    smallest = find(array_shapes(min_dims, min_dims + dim_range,
+                                 min_side, min_side + side_range),
+                    lambda x: True)
+    assert len(smallest) == min_dims and all(k == min_side for k in smallest)
+
+
+@given(scalar_dtypes())
+def test_can_generate_scalar_dtypes(dtype):
+    assert isinstance(dtype, np.dtype)
+
+
+@given(nested_dtypes())
+def test_can_generate_compound_dtypes(dtype):
+    assert isinstance(dtype, np.dtype)
+
+
+@given(nested_dtypes())
+def test_infer_strategy_from_dtype(dtype):
+    # Given a dtype
+    assert isinstance(dtype, np.dtype)
+    # We can infer a strategy
+    strat = from_dtype(dtype)
+    assert isinstance(strat, SearchStrategy)
+    # And use it to fill an array of that dtype
+    assume(10 * dtype.itemsize <= settings().buffer_size)
+    arrays(dtype, 10, strat)
```

Comment:
It would look something like this:

```python

@given(nested_dtypes(), st.data())
 def test_infer_strategy_from_dtype(dtype, data):
     # Given a dtype
     assert isinstance(dtype, np.dtype)
     # We can infer a strategy
     strat = from_dtype(dtype)
     assert isinstance(strat, SearchStrategy)
     # And use it to fill an array of that dtype
     assume(10 * dtype.itemsize <= settings().buffer_size)
     data.draw(arrays(dtype, 10, strat))
```